### PR TITLE
Modify radio button UI

### PR DESCRIPTION
**Pull request by ElishaSamPeterPrabhu**
## Description

RadioButton UI is updated according to Figma 1.2.19 and story book of radio button is updated as well

References #1877

Preview: https://icy-ground-0374a1310-1895.centralus.1.azurestaticapps.net/?path=/docs/components-radio-group--medium#radio-group

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## How Has This Been Tested?
Unit tests are done to check the size prop

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: stencil-workspace/src/components/modus-radio-group/modus-radio-button.vars.scss [hunk 1]
```diff
@@ -8,3 +8,6 @@ $modus-radio-btn-hover-border-color: var(--modus-radio-btn-hover-border-color, #
 
 // Checked state
 $modus-radio-btn-selected-border-color: var(--modus-radio-btn-selected-border-color, #217cbb) !default;
+
+// Focus state
+$modus-radio-btn-focus-box-shadow-color: var(--modus-checkbox-focus-box-shadow-color, rgba(#217cbb, 0.6)) !default;
```
Comment (coliff) [inline stencil-workspace/src/components/modus-radio-group/modus-radio-button.vars.scss:13]: `rgba(#217cbb, 0.6)` is invalid CSS. The rgba() function in CSS expects three arguments for the red, green, and blue color values, and a fourth argument for the alpha channel.

Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/src/components/modus-radio-group/modus-radio-button.vars.scss:13]: Modified to 
```
// Focus state
$modus-radio-btn-focus-box-shadow-color: var(--modus-checkbox-focus-box-shadow-color, #019aeb) !default;
```

File: stencil-workspace/src/components/modus-radio-group/modus-radio-button.tsx [hunk 2]
```diff
@@ -16,18 +17,22 @@ interface ModusRadioButtonProps {
...
-  <div
-    class={`modus-radio-button ${props.disabled ? 'disabled' : ''}`}
-    onClick={() => props.handleButtonClick(props.id)}
-    onKeyDown={(event) => props.handleKeydown(event, props.id)}
-    tabIndex={0}>
-    <div class="radio">
-      <input checked={props.checked} disabled={props.disabled} name={props.name} type="radio"></input>
-      <span class="checkmark"></span>
-      <label>{props.label}</label>
+  console.log('props', props.size),
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-radio-group/modus-radio-button.tsx:24]: remove console log

File: stencil-workspace/src/components/modus-radio-group/modus-radio-group.tsx [hunk 3]
```diff
@@ -56,6 +59,7 @@ export class ModusRadioGroup {
         {this.radioButtons.map((radioButton) => {
           return (
             <ModusRadioButton
+              size={radioButton.size}
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-radio-group/modus-radio-group.tsx:62]: each button in the radio group must not have a separate size property

File: stencil-workspace/storybook/stories/components/modus-radio-group/modus-radio-group-storybook-docs.mdx [hunk 4]
```diff
@@ -35,6 +36,7 @@ A new TypeScript typing has been provided named RadioButton defined as:
       id: '1',
       checked: true,
       label: 'Radio 2',
+      size:'small'
```
Comment (msankaran0712) [inline stencil-workspace/storybook/stories/components/modus-radio-group/modus-radio-group-storybook-docs.mdx:39]: each radio button in the group must not have a separate size

File: stencil-workspace/storybook/stories/components/modus-radio-group/modus-radio-group-storybook-docs.mdx [hunk 5]
```diff
@@ -43,6 +45,35 @@ A new TypeScript typing has been provided named RadioButton defined as:
...
+    {
+      id: '0',
+      label: 'Radio 1',
+      size:'small'
+    },
+    {
+      id: '1',
+      checked: true,
+      label: 'Radio 2',
+      size:'small'
```
Comment (msankaran0712) [inline stencil-workspace/storybook/stories/components/modus-radio-group/modus-radio-group-storybook-docs.mdx:67]: same, please remove size from the radio buttons in a group. Size must be set on the group component.

File: stencil-workspace/storybook/stories/components/modus-radio-group/modus-radio-group.stories.tsx [hunk 6]
```diff
@@ -47,3 +52,29 @@ const setRadioGroup = () => {
...
+      {
+        id: '0',
+        label: 'Radio 1',
+        size:'small'
+      },
+      {
+        checked: true,
+        id: '1',
+        label: 'Radio 2',
+        size:'small'
```
Comment (msankaran0712) [inline stencil-workspace/storybook/stories/components/modus-radio-group/modus-radio-group.stories.tsx:69]: remove size from each button

**Review by rthanga1 — DISMISSED** [review]

File: stencil-workspace/src/components/modus-radio-group/modus-radio-button.tsx (hunk 2 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/src/components/modus-radio-group/modus-radio-button.tsx:24]: Removed

File: stencil-workspace/storybook/stories/components/modus-radio-group/modus-radio-group.stories.tsx (hunk 6 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/storybook/stories/components/modus-radio-group/modus-radio-group.stories.tsx:69]: Removed

File: stencil-workspace/storybook/stories/components/modus-radio-group/modus-radio-group-storybook-docs.mdx (hunk 5 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/storybook/stories/components/modus-radio-group/modus-radio-group-storybook-docs.mdx:67]: Removed

File: stencil-workspace/storybook/stories/components/modus-radio-group/modus-radio-group-storybook-docs.mdx (hunk 4 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/storybook/stories/components/modus-radio-group/modus-radio-group-storybook-docs.mdx:39]: changed now its a group property

File: stencil-workspace/src/components/modus-radio-group/modus-radio-group.tsx (hunk 3 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/src/components/modus-radio-group/modus-radio-group.tsx:62]: Changed now its a group property

**Review by msankaran0712 — CHANGES_REQUESTED** [review]
@ElishaSamPeterPrabhu On both light and dark themes, the focus state does not look exactly like in the Figma. Also, the inner dot looks slightly not centered (only a minor observation, no need to fix if it is difficult)

Storybook:
image
 
Figma:
image

File: stencil-workspace/src/components/modus-radio-group/modus-radio-group.scss [hunk 7]
```diff
@@ -71,6 +89,12 @@ div.modus-radio-group {
     border-color: $modus-radio-btn-hover-border-color;
   }
 
+  &:focus .radio .checkmark {
+    border-color: $modus-radio-btn-hover-border-color;
+    box-shadow: 0 0 0 0.2rem $modus-radio-btn-focus-box-shadow-color;
+    outline: 2px solid white;
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-radio-group/modus-radio-group.scss:95]: White outline color is not expected on dark theme I guess

**Comment by ElishaSamPeterPrabhu** [comment]
> (Optional) If possible can you please fix the bug which is about selecting the radio button when clicking outside RadioButtons [ RadioButtons ](https://private-user-images.githubusercontent.com/50131391/287720896-36e83b6e-42b7-44b3-8583-38c707bbb797.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE3NzA2ODcsIm5iZiI6MTcwMTc3MDM4NywicGF0aCI6Ii81MDEzMTM5MS8yODc3MjA4OTYtMzZlODNiNmUtNDJiNy00NGIzLTg1ODMtMzhjNzA3YmJiNzk3LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA1VDA5NTk0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZhYzIwMGFkM2I5YjVlOTI0NDI0YzllY2I2Y2JiZGM2NDlmODc3MjNmMjZjMDZjYjU1YWFlNTE1MTI5MGQ1YzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ESvpGRTxKXk3QUlhED2KJy6tz2t1VLj-OHV7PWoT7_I) [ ](https://private-user-images.githubusercontent.com/50131391/287720896-36e83b6e-42b7-44b3-8583-38c707bbb797.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE3NzA2ODcsIm5iZiI6MTcwMTc3MDM4NywicGF0aCI6Ii81MDEzMTM5MS8yODc3MjA4OTYtMzZlODNiNmUtNDJiNy00NGIzLTg1ODMtMzhjNzA3YmJiNzk3LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA1VDA5NTk0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZhYzIwMGFkM2I5YjVlOTI0NDI0YzllY2I2Y2JiZGM2NDlmODc3MjNmMjZjMDZjYjU1YWFlNTE1MTI5MGQ1YzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ESvpGRTxKXk3QUlhED2KJy6tz2t1VLj-OHV7PWoT7_I)

Fixed

File: stencil-workspace/src/components/modus-radio-group/modus-radio-button.tsx [hunk 8]
```diff
@@ -16,18 +16,21 @@ interface ModusRadioButtonProps {
...
   <div
     class={`modus-radio-button ${props.disabled ? 'disabled' : ''}`}
     onClick={() => props.handleButtonClick(props.id)}
     onKeyDown={(event) => props.handleKeydown(event, props.id)}
     tabIndex={0}>
     <div class="radio">
       <input checked={props.checked} disabled={props.disabled} name={props.name} type="radio"></input>
-      <span class="checkmark"></span>
-      <label>{props.label}</label>
+      <span class={`checkmark ${props.size}`}></span>
```
[the remark this answers is not in the document]
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/src/components/modus-radio-group/modus-radio-button.tsx]: Moved to parent div

File: stencil-workspace/src/components/modus-radio-group/modus-radio-group.scss (hunk 7 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/src/components/modus-radio-group/modus-radio-group.scss]: Modified approach

**Comment by coliff** [comment]
Radio buttons are too big. The white inner circle should be 12px height/width, but this PR has it at 14px. Can you reduce by 2px?

**Comment by coliff** [comment]
Inner circle isn't centered:
image

Please squash commits and mark as ready for review when that's done - thanks!

**Review by coliff — APPROVED** [review]

**Comment by cjwinsor** [comment]
@ElishaSamPeterPrabhu I see the index.html was updated. Is it possible to get a correction PR to clear it out?

**Pull request merged**